### PR TITLE
adding fields on jobs offers

### PR DESCRIPTION
**Pull request by eerison**
Issue https://github.com/kdti/kdti-server/issues/52

### CODE REVIEW

**Comment by Luitame** [comment]
Does it close any open issues?

File: src/Entity/JobOffer.php [hunk 1]
```diff
@@ -19,6 +19,9 @@ class JobOffer
     const STATUS_PENDING_REVIEW = 'PENDING_REVIEW';
     const STATUS_APPROVED = 'APPROVED';
 
+    const HIRING_TYPE_CLT = 'CLT';
```
Comment (rtio) [inline src/Entity/JobOffer.php]: It would be nice if we use Enum instead of constants. Will approve BTW

File: src/Entity/JobOffer.php [hunk 2]
```diff
@@ -83,7 +86,7 @@ class JobOffer
     /**
      * @ORM\Column(type="string", length=14)
      * 
-     * @Groups({"admin"})
+     * @Groups({"admin", "detail", "list"})
```
Comment (rtio) [inline src/Entity/JobOffer.php]: I don't see why we need to show the status in detail and list groups. The company that the job belongs to will see the status in the admin group. BTW, out of scope.

File: src/Entity/JobOffer.php [hunk 3]
```diff
@@ -98,16 +101,24 @@ class JobOffer
...
      * @ORM\Column(type="datetime")
+     *
      */
     private $updatedAt;
 
     /**
      * @ORM\Column(type="datetime", nullable=true)
      *
-     * @Groups({"admin", "detail"})
+     * @Groups({"admin", "detail", "list"})
```
Comment (rtio) [inline src/Entity/JobOffer.php]: Out of scope.

File: src/Entity/JobOffer.php [hunk 4]
```diff
@@ -98,16 +101,24 @@ class JobOffer
...
      */
     private $publishedAt;
 
+    /**
+     * @ORM\Column(type="string", length=3)
+     *
+     * @Groups({"admin", "detail", "list"})
```
Comment (rtio) [inline src/Entity/JobOffer.php]: Are we sure that we are going to show this field in all serializer groups?

File: tests/Feature/ListJobOffersTest.php [hunk 5]
```diff
@@ -36,9 +36,11 @@ public function test_should_list_only_approved_job_offers(): void
             'minimumSalary' => 2750,
             'maximumSalary' => 3000,
             'status' => JobOffer::STATUS_APPROVED,
+            'hiring_type' => JobOffer::HIRING_TYPE_CLT,
```
Comment (rtio) [inline tests/Feature/ListJobOffersTest.php]: Why is this field snake_case? Is it passing on the tests?

File: tests/Feature/DisplaySingleJobOfferTest.php [hunk 6]
```diff
@@ -38,6 +38,7 @@ public function test_display_a_single_job_offer(): void
             'minimumSalary' => 4000,
             'maximumSalary' => 4500,
             'status' => JobOffer::STATUS_APPROVED,
+            'hiring_type' => JobOffer::HIRING_TYPE_CLT,
```
Comment (rtio) [inline tests/Feature/DisplaySingleJobOfferTest.php]: Why is this field snake_case? Is it passing on the tests?

File: tests/Feature/DisplaySingleJobOfferTest.php [hunk 7]
```diff
@@ -59,6 +60,7 @@ public function test_displaying_non_approved_job_offers_causes_404(): void
             'minimumSalary' => 3000,
             'maximumSalary' => 3200,
             'status' => JobOffer::STATUS_PENDING_REVIEW,
+            'hiring_type' => JobOffer::HIRING_TYPE_PJ,
```
Comment (rtio) [inline tests/Feature/DisplaySingleJobOfferTest.php]: Why is this field snake_case? Is it passing on the tests?

File: tests/Feature/DisplaySingleJobOfferTest.php [hunk 8]
```diff
@@ -76,8 +78,9 @@ public function test_display_by_slug_a_single_job_offer(): void
             'seniorityLevel' => 'Senior',
             'minimumSalary' => 1121,
             'maximumSalary' => 2128,
-            'status' => JobOffer::STATUS_APPROVED,
             'publishedAt' => new DateTime('2019-01-01'),
+            'status' => JobOffer::STATUS_APPROVED,
+            'hiring_type' => JobOffer::HIRING_TYPE_CLT,
```
Comment (rtio) [inline tests/Feature/DisplaySingleJobOfferTest.php]: Why is this field snake_case? Is it passing on the tests?

File: tests/Feature/ListJobOffersTest.php [hunk 9]
```diff
@@ -36,9 +36,11 @@ public function test_should_list_only_approved_job_offers(): void
...
         ]);
         $this->factory->create(JobOffer::class, [
             'status' => JobOffer::STATUS_PENDING_REVIEW,
+            'hiring_type' => JobOffer::HIRING_TYPE_CLT,
```
Comment (rtio) [inline tests/Feature/ListJobOffersTest.php]: Why is this field snake_case? Is it filling the factory properly?

File: tests/Unit/Entity/CompanyTest.php [hunk 10]
```diff
@@ -13,7 +14,7 @@ class CompanyTest extends TestCase
     public function test_string_representation(): void
     {
         $company = $this->factory->create(Company::class, [
-            'name' => 'Dunder Mifflin',
+            'name' => 'Dunder Mifflin'
```
Comment (rtio) [inline tests/Unit/Entity/CompanyTest.php]:
```suggestion
            'name' => 'Dunder Mifflin',
```

File: tests/Feature/DisplaySingleJobOfferTest.php (hunk 6 above)
Comment (eerison) [reply tests/Feature/DisplaySingleJobOfferTest.php]: yes

PHPUnit 7.5.18 by Sebastian Bergmann and contributors.

Testing Project Test Suite
............................                                      28 / 28 (100%)

Time: 25.89 seconds, Memory: 62.50 MB

OK (28 tests, 42 assertions)


But I fixed

File: src/Entity/JobOffer.php (hunk 1 above)
Comment (eerison) [reply src/Entity/JobOffer.php]: Sorry I did similar the `status` field

Comment (rtio) [reply src/Entity/JobOffer.php]: Ok. Will put it to be done in another moment

File: tests/Feature/__snapshots__/DisplaySingleJobOfferTest__test_display_a_single_job_offer.json [hunk 11]
```diff
@@ -15,5 +15,7 @@
     "minimumSalary": 4000,
     "maximumSalary": 4500,
     "publishedAt": "2019-01-01T00:00:00+0000",
+    "status": "APPROVED",
```
Comment (rtio) [inline tests/Feature/__snapshots__/DisplaySingleJobOfferTest__test_display_a_single_job_offer.json]: status shouldn't appear here

File: tests/Feature/__snapshots__/DisplaySingleJobOfferTest__test_display_by_slug_a_single_job_offer.json [hunk 12]
```diff
@@ -15,5 +15,7 @@
     "minimumSalary": 1121,
     "maximumSalary": 2128,
     "publishedAt": "2019-01-01T00:00:00+0000",
+    "status": "APPROVED",
```
Comment (rtio) [inline tests/Feature/__snapshots__/DisplaySingleJobOfferTest__test_display_by_slug_a_single_job_offer.json]: status shouldn't appear here

File: tests/Feature/__snapshots__/ListJobOffersTest__test_should_list_only_approved_job_offers.json [hunk 13]
```diff
@@ -10,6 +10,9 @@
         },
         "seniorityLevel": "Senior",
         "minimumSalary": 2750,
-        "maximumSalary": 3000
+        "maximumSalary": 3000,
+        "status": "APPROVED",
```
Comment (rtio) [inline tests/Feature/__snapshots__/ListJobOffersTest__test_should_list_only_approved_job_offers.json]: status shouldn't appear here

File: src/Entity/JobOffer.php (hunk 3 above)
Comment (eerison) [reply src/Entity/JobOffer.php]: WhatsApp Image 2019-12-19 at 17 01 14

Edenilson is using here (public list)

**Comment by rtio** [comment]
Great!!

**Pull request merged**